### DISCUSSION TITLE
HTTP tests - multipart request are in request.files() instead of request.file()

### DISCUSSION THREAD

**Discussion by mortenko** [adonisjs/core#1362] [Help]
In the section  **Multipart Requests** of documentation **HTTP test**, you can send multiple files using attach method and catch these files in controller using **request.files()** (request.file() gives me **null**) but in my real world app Im sending just one file over the internet and catching it in **request.file()**. There is no information about plural form  **request.files()**. 

Is there any possibility to mock http client and attach single file ? 

```js
await client
  .post('posts')
  .field('title', 'Adonis 101')
  .attach('cover_image', Helpers.tmpPath('cover-image.jpg')) // this can be catch only with request.files()
  .end()
```

**Comment by thetutlage** [comment]
Ideally this should be the case. You should be able to access the file as `request.file('cover_image')`. Lemme try to reproduce it myself and get back to you.
